lib/stdlib/test/erl_lint_SUITE.erl

```diff
@@ -4448,6 +4449,46 @@ inline_nifs(Config) ->
            {warnings,[{2,erl_lint,nif_inline}]}}],
     [] = run(Config, Ts).
 
+compile_options(Config) ->
+    WarnMissSpecOkay
+        = {warn_missing_spec_ok,
+           <<"-export([external/0]).
+              -spec external() -> ok. external() -> internal().
+              internal() -> ok.">>,
+           [warn_missing_spec],
+           []},
+
+    WarnMissSpecNotOkay
+        = {warn_missing_spec_error,
+           <<"-export([external/0]).
+              external() -> internal().
+              -spec internal() -> ok. internal() -> ok.">>,
+           [warn_missing_spec],
+           {warnings, [{2, erl_lint, {missing_spec, {external, 0}}}]}},
+
+    WarnMissSpecAllOkay
+        = {warn_missing_spec_all_ok,
+           <<"-export([external/0]).
+              -spec external() -> ok. external() -> internal().
+              -spec internal() -> ok. internal() -> ok.">>,
+           [warn_missing_spec_all],
+           []},
+
+    WarnMissSpecAllNotOkay
+        = {warn_missing_spec_all_error,
+           <<"-export([external/0]).
+              external() -> internal().
+              -spec internal() -> ok. internal() -> ok.">>,
+           [warn_missing_spec_all],
+           {warnings, [{2, erl_lint, {missing_spec, {external, 0}}}]}},
```

Comment:
You are not testing that there will be a warning for a local function without a spec. Similar to the first one, these tests can be combined. Note that you can also reuse the identical source code (i.e. assign the binary with the source to a variable in the beginning of the function and use it in the tests for both `warn_missing_spec` and `warn_missing_spec_all`).

Reply:
Hm... I want to test the following:
1. no error for `warn_missing_spec_all` when everything's OK (this includes internal functions),
2. error for `warn_missing_spec_all` when there's something missing (this includes internal functions).

Reply:
In the second test (`WarnMissSpecNotOkay`), I'm making sure that `warn_missing_spec_all` also cares about external specs.

Reply:
> Note that you can also reuse the identical source code

I can, and will, since this appears to be your general guidance. I just tend to make tests as atomic as possible so one output doesn't influence another and changing one "datum" doesn't affect more than one test.